### PR TITLE
DSDL documentation hosting

### PR DESCRIPTION
**Pull request by bbworld1**
image
This PR introduces support for hosting of Nunavut autogenerated documentation.

Blocked by https://github.com/UAVCAN/nunavut/pull/202.

### CODE REVIEW

File: nunaserver/nunaserver/generator.py [hunk 1]
```diff
@@ -169,24 +171,53 @@ def generate_dsdl(
...
-    )
+    if target_lang == "html":
+        # Upload generated files
+        for file in out_dir.glob("**/*"):
+            if file.is_file():
+                rel_path = doc_url / file.relative_to(out_dir)
+                try:
+                    if storage.stat_object("docs", f"{rel_path}"):
+                        raise RuntimeError("Specified doc URL is already taken")
+                except minio.error.S3Error:
```
Comment (pavel-kirienko) [inline nunaserver/nunaserver/generator.py:182]: Some logging is in order I suppose

**Review by pavel-kirienko — COMMENTED** [review]
What is the persistence policy of the generated docs? How long the results are kept on the server?

File: nunaserver/nunaserver/generator.py (hunk 1 above)
Comment (bbworld1) [reply nunaserver/nunaserver/generator.py:182]: That's something we have to decide. I don't have a delete policy right now; I can set it to 30 days, same as how long it keeps results.

**Review by pavel-kirienko — APPROVED** [review]
Let's have the doc lifecycle management issue documented pls as discussed at the call today.

**Comment by pavel-kirienko** [comment]
<sub>(@bbworld1 nudge?)</sub>

**Comment by bbworld1** [comment]
Will do. I'll submit a separate issue request; the 30 day delete policy should work fine in the meantime.

Can you merge this?

**Comment by pavel-kirienko** [comment]
Sure thing. I thought you are going to do it since you are maintaining this project, no need to wait for me.

**Pull request merged**